src/node/mod.rs

```diff
@@ -321,6 +320,30 @@ pub trait AudioNode {
     fn set_channel_count(&self, v: usize) {
         self.channel_config().set_count(v)
     }
+
+    /// Register callback to run when an unhandled exception occurs in the audio processor.
+    ///
+    /// Note that once a unhandled exception is thrown, the processor will output silence throughout its lifetime.
+    ///
+    /// Only a single event handler is active at any time. Calling this method multiple times will
+    /// override the previous event handler.
+    fn set_onprocessorerror(&self, callback: Box<dyn FnOnce(ErrorEvent) + Send + 'static>) {
```

Comment:
The AudioNode trait cannot take any generic arguments in its method. Compiler will complain `this trait cannot be made into an object`. Hence we need the user to box it already to erase the generic fn type

Reply:
I just wonder if we just not generalize this API to all events then for consistency?

Reply:
Tough choice, I dislike that we need to ask users to box the closure. It really is a lot nicer to just pass the closure in raw. To the user it will not be very clear why they need to box it, anyway
I think I will merge this version, but feel free to open the discussion again if you think the consistency is more important